### ISSUE TITLE
Adb push error: 

### ISSUE THREAD

**Issue by the-one-who-blocks** [ubports/ubports-installer#582]
*Automatically generated error report* 
UBports Installer Version: 0.1.21-beta 
Device: Nexus 4
Package: unknown
Operating System: Ubuntu Linux 18.04 bionic x64 
NodeJS version: v8.2.1 

Error log: http://paste.ubuntu.com/p/r9ctCVw7p7/ 

ubports brings up to error window saying the installer is not the most up to date version, unless there is another one hiding 0.1.21 is the newest I can find 

UBports recovery has been installed but the files for the ROM won't push

**Comment by cobainmo** [comment]
I have the same issue. Tried Windows, Mac and Ubuntu. Installer did not work at all. Finally it worked on mac with VirtualBox and ubuntu but only fastboot part. When it boot to recovery mode, it just output errors about adb push

**Comment by advocatux** [comment]
@cobainmo that sounds like a bad usb cable to me. Have you tried with a different one?

**Comment by cobainmo** [comment]
I solved the issue finally. @advocatux it was not cable issue.  In the VirtualBox Settings-> Ports -> USB - USB Device Filters i need to add USB device to filter in both state, fastboot and recovery or system boot when adb is avialable. Also if you do not torn on USB debugging and allow computer you will get unauthorized message in adb devices list.

This took a lot of nerves 😄  I wrote instructions that helped me here: https://gist.github.com/cobainmo/f8532eec9a3d751d9c9adc9703704f1b
